### PR TITLE
Migrate to PCRE2

### PR DESCRIPTION
**Pull request by vpodzime**
PCRE has been deprecated for long enough.

Ticket: ENT-10629
Changelog: CFEngine now uses PCRE2 for regular expressions

Merge together:
https://github.com/cfengine/core/pull/5391
https://github.com/cfengine/buildscripts/pull/1343
https://github.com/cfengine/enterprise/pull/779
https://github.com/cfengine/nova/pull/2160

### CODE REVIEW

**Comment by cf-bottom** [comment]
Thank you for submitting a pull request! Maybe @craigcomstock can review this?

**Review by craigcomstock — COMMENTED** [review]
looks good :)

File: cf-execd/exec-config.c [hunk 1]
```diff
@@ -54,30 +54,44 @@ static char *GetIpAddresses(const EvalContext *ctx)
...
 {
-    const char *errorstr;
-    int erroffset;
-    pcre *rx = pcre_compile(str,
-                            PCRE_MULTILINE | PCRE_DOTALL | PCRE_ANCHORED,
-                            &errorstr, &erroffset, NULL);
-    if (!rx)
+    int err_code;
+    size_t err_offset;
+    pcre2_code *regex = pcre2_compile((PCRE2_SPTR) pattern, PCRE2_ZERO_TERMINATED,
```
[the remark this answers is not in the document]
Comment (vpodzime) [reply cf-execd/exec-config.c:66]: yup

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/10136/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/10136/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-10136/

**Comment by vpodzime** [comment]
@cf-bottom jenkins with exotics, please

**Comment by vpodzime** [comment]
Build with the `Regex` type (no tests): [Build Status](https://ci.cfengine.com/job/pr-pipeline/10150/)

**Review by larsewi — APPROVED** [review]
🚀

File: libpromises/matching.c [hunk 2]
```diff
@@ -37,47 +37,45 @@
...
+                             0, 0, match_data, NULL);
+    /* pcre2_match() returns the highest capture group number + 1, i.e. 1 means
+     * a match with 0 capture groups. 0 means the vector of offsets is small,
+     * negative numbers are errors (incl. no match). */
+    if (result > 0)
     {
-        for (i = 1; i < rc; i++)        /* make backref vars $(1),$(2) etc */
+        size_t *ovector = pcre2_get_ovector_pointer(match_data);
+        /* ovector[0] and ovector[1] are for the start and end of the whole
+         * match, the capture groups follow in [1] and [2], etc. */
```
Comment (craigcomstock) [inline libpromises/matching.c:55]: did you mean [2] and [3]? since ovector[1] is the end of the whole match?

**Review by craigcomstock — APPROVED** [review]

File: libpromises/matching.c (hunk 2 above)
Comment (vpodzime) [reply libpromises/matching.c:55]: https://github.com/cfengine/core/pull/5399

**Pull request merged**